doc/nrf/ug_radio_coex.rst

```diff
@@ -0,0 +1,194 @@
+
+In order to mitigate coexistence issues and improve coexistence performance a Packet Traffic Arbiter (PTA) is used. Both SR and WiFi request access to RF
+and PTA grants/denies that access.
+
+This guide describes how to add WiFi Coexistence (CX) support to your SR application in |NCS|. In order to do it, both
+:ref:`Software support` and :ref:`Hardware description` must be added.
+
+Software support
+****************
+
+If your application uses SR protocols and requires coexistence with WiFi, you must enable :ref:`nrfxlib:mpsl_cx` in the
+:ref:`nrfxlib:mpsl` (MPSL) library.
+
+CX with MPSL
+============
+
+The MPSL library provides an implementation for the WiFi Coexistens module for SR applications.
+To use this implementation, your application must use a protocol driver that enables the WiFi Coexistence (CX) feature.
+The library provides multiprotocol support, but you can also use it in applications that require only one protocol.
+To avoid conflicts, check the protocol documentation to see if the protocol uses CX support provided by MPSL.
+
+.. note::
+   Currently, the following protocols use CX support provided by MPSL:
+
+   * :ref:`ug_thread`
+   * :ref:`ug_zigbee`
+   * :ref:`ug_ble_controller`
+   * :ref:`ug_multiprotocol_support`
+
+.. _ug_radio_cx_requirements:
+
+Enabling MPSL
+-------------
+
+Before you add the devicetree node in your application, complete the following steps:
+
+   The MPSL library provides API to handle signals coming from PTA and controlling requests send to PTA.
+   See :ref:`nrfxlib:mpsl_lib` in the nrfxlib documentation for details.
+
+1. Add support for the MPSL library in your application.
+2. Enable support for MPSL implementation in |NCS| by setting the :kconfig:`CONFIG_MPSL` Kconfig option to ``y``.
+
+Enabling CX
+-------------
+
+In order to enable CX:
+
+1. Set the :kconfig:`CONFIG_MPSL_CX` Kconfig option to ``y``.
+2. Enable support for specific Coex implementation. The currently supported implementations are "generic 3-pin interface" and "thread", for
+   details see :ref:`WiFi Coex implementations`.
+
+Hardware description
+********************
+
+The |NCS| provides a wrapper that configures WiFi Coexistence based on devicetree (DTS) and Kconfig information.
+
+To enable CX support for any for the currently supported CX implementation, you must add an ``nrf_radio_coex`` node in the devicetree file. The
+node can also be provided by the devicetree file of the target development kit or by an overlay file. See :ref:`zephyr:dt-guide` for more
+information about the DTS data structure, and :ref:`zephyr:dt_vs_kconfig` for information about differences between DTS and Kconfig.
+
+Note: When adding custom user implementation a custom dts node can be added, if the ``nrf_radio_coex`` node is unsuitable.
+
+The ``nrf_radio_coex`` dts represents a situation in which a 3 wire interface to the PTA is provided, as shown in the image below. However, the
```

Comment:
This information is a bit misleading. `nrf_radio_coex` doesn't define any specific interface to the PTA. It's a node label that the DTS file and the C implementation "agree on". That's why the following sentence you wrote is perfectly true:

> To enable CX support for any for the currently supported CX implementation, you must add an ``nrf_radio_coex`` node in the devicetree file.

as well as this one:

> When adding custom user implementation a custom dts node can be added, if the ``nrf_radio_coex`` node is unsuitable.

Please have a look at this sentence and remove/rework it so that it's not misleading anymore :)

Reply:
I must say I don't quite understand what is the misleading part here, I've tried to emphasize that it's up to the implementation to decide what to do with the information provided by the dts node, but I tried to make it better - please see if the change I introduced made it less misleading.

Reply:
OK, let me clarify.

Essentially, there are three pieces of software that need to fit together for a piece of hardware such as a Coexistence interface to be controllable from C code:
* binding, which describes a "type" of device or hardware; personally, I think of it as an equivalent of a complex `typedef`. For example, the following binding: https://github.com/nrfconnect/sdk-nrf/blob/main/dts/bindings/radio_coex/generic-radio-coex-three-wire.yaml describes a generic three wire Coexistence interface.
* dts file, which defines an instance of a hardware of a given type. This is the piece of code you're describing in this section of the documentation. Its job is to state that a hardware of a given type (i.e. with its `compatible` property equal to that of the binding, e.g. `generic-radio-coex-three-wire`) is present on a given platform/board/target.
* C code, that combines the information about the "type" from the binding with the information about the "instance" from the dts file to actually control that given piece of hardware.

The current Coexistence implementations in C expect that the "instance" of Coexistence interface they want to control is labelled `nrf_radio_coex`:
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/mpsl/cx/thread/mpsl_cx_thread.c#L28-L33,
https://github.com/nrfconnect/sdk-nrf/blob/main/subsys/mpsl/cx/generic_3pin/mpsl_cx_generic_3pin.c#L41-L46.

Therefore, is a user wishes to use the generic 3 pin plugin provided in sdk-nrf, they need to add a devicetree node compatible with `generic-rado-coex-three-wire` binding that is called `nrf_radio_coex`. Otherwise, the C code will not know how to control it or it will not recognize the "instance" at all. Similarly, if a user wishes to implement their own plugin, they need to add a devicetree node compatible with their custom binding and name it whatever they like. The sentences I cited earlier convey exactly that information.

However, `nrf_radio_coex` doesn't represent _anything at all_. It's just a string that the dts file and the C code implicitly agree on to "recognize" each other. It could be `node_where_magic_happens` instead and the solution would work exactly the same.

That's why I wrote that the information is misleading :) I'd suggest to rework this piece of documentation into something like:

> Currently supported Coexistence interfaces are sample implementations of generic three wire interface: https://github.com/nrfconnect/sdk-nrf/blob/main/dts/bindings/radio_coex/generic-radio-coex-three-wire.yaml. They all follow the design outlined by the picture below. However, the role of each of the pins is dependent on the WiFi Coexistence implementation.

If the solution feels complicated to you, it's justified - it **is** complicated. That's why the generic pin forwarder: https://github.com/zephyrproject-rtos/zephyr/pull/43047 was introduced. It should help to simplify that design.